java/ql/lib/semmle/code/java/frameworks/android/Intent.qll

```diff
@@ -57,6 +64,17 @@ class AndroidReceiveIntentMethod extends Method {
   }
 }
 
+/**
+ * The method `Service.onStart`, `onStartCommand`,
+ * `onBind`, `onRebind`, `onUnbind`, or `onTaskRemoved`.
+ */
+class AndroidServiceIntentMethod extends Method {
+  AndroidServiceIntentMethod() {
+    this.getName().matches(["onStart%", "on%ind", "onTaskRemoved"]) and
```

Comment:
I'd suggest this change:

```suggestion
    this.hasName(["onStart%", "onBind", "onRebind", "onUnbind", "onTaskRemoved"]) and
```

Your version is shorter, but in general it's better to use `hasName` instead of `getName` for performance reasons, and it's more readable this way.

Reply:
Post-merge drive-by comment - I just thought I'd clarify something here. The predicates `getName` and `hasName` are completely equivalent - they both exist since sometimes one is more convenient than the other. The potential performance difference is between equality to a set of given names and using `matches`. The former means that the join-orderer is able to start with the small set of names and look up those methods - this is generally efficient as the amount of work is proportional to the list of names. Whereas the latter approach of using `matches` means that we have to start with _all_ methods (potentially filtered by other conjuncts) and filter those based on whether they match a patten - this is generally less efficient as the amount of work tends to be proportional to the number of methods times the number of patterns.